okhttp-logging-interceptor/src/main/kotlin/okhttp3/logging/HttpLoggingInterceptor.kt

```diff
@@ -352,16 +476,107 @@ class HttpLoggingInterceptor
       logger.log(headers.name(i) + ": " + value)
     }
 
-    private fun bodyHasUnknownEncoding(headers: Headers): Boolean {
-      val contentEncoding = headers["Content-Encoding"] ?: return false
-      return !contentEncoding.equals("identity", ignoreCase = true) &&
-        !contentEncoding.equals("gzip", ignoreCase = true)
+    private fun findCompressionAlgorithm(
+      headers: Headers,
+      decompressionAlgorithms: List<DecompressionAlgorithm>,
+    ): DecompressionAlgorithm? {
+      val contentEncoding = headers["Content-Encoding"] ?: return Identity
+
+      if (contentEncoding.equals("identity", ignoreCase = true)) return Identity
+
+      return decompressionAlgorithms.find { contentEncoding.equals(it.encoding, ignoreCase = true) }
     }
 
-    private fun bodyIsStreaming(response: Response): Boolean {
+    object Identity : DecompressionAlgorithm {
+      override val encoding: String
+        get() = "identity"
+
+      override fun decompress(compressedSource: BufferedSource): Source = compressedSource
+    }
+
+    private fun bodyIsEventStream(response: Response): Boolean {
       val contentType = response.body.contentType()
       return contentType != null && contentType.type == "text" && contentType.subtype == "event-stream"
     }
 
     companion object
   }
+
+private class StreamingRequestBody(
```

Comment:
*every* request body is streaming; this is just streaming the body to the logs instead of buffering it all